### PR TITLE
Make ActiveWidgetStore clear persistent widgets

### PR DESCRIPTION
**Pull request by dbkr**
ActiveWidgetStore is now reponsible for removing the current
persistent widget from the store if it's been removed from whatever
room it was in. As per comment, this leaves us with the store updating
itself in this case but in all other cases, views call setters on the
store to update its state. We should make it so the store keeps itself
up to date and views aren't responsible for keeping the store up to date.

The store now emits events so it can notify PersistentApp when it changes.

Fixes vector-im/riot-web#7076
Based on https://github.com/matrix-org/matrix-react-sdk/pull/2081

### CODE REVIEW

File: src/components/views/elements/AppTile.js [hunk 1]
```diff
@@ -161,14 +161,18 @@ export default class AppTile extends React.Component {
...
-            const PersistedElement = sdk.getComponent("elements.PersistedElement");
-            PersistedElement.destroyElement(this._persistKey);
-            ActiveWidgetStore.delWidgetMessaging(this.props.id);
-            ActiveWidgetStore.delWidgetCapabilities(this.props.id);
-            ActiveWidgetStore.delRoomId(this.props.id);
+            this._destroyPersistentWidget();
         }
     }
 
+    _destroyPersistentWidget() {
```
Comment (bwindels) [inline src/components/views/elements/AppTile.js]: seems duplicated with ActiveWidgetStore::onRoomStateEvents. Have a single public method in ActiveWidgetStore that does this?

File: src/Lifecycle.js [hunk 2]
```diff
@@ -488,6 +490,7 @@ export function stopMatrixClient() {
     Notifier.stop();
     UserActivity.stop();
     Presence.stop();
+    ActiveWidgetStore.start();
```
Comment (bwindels) [inline src/Lifecycle.js]: this should probably be `stop()`

File: src/components/views/elements/AppTile.js (hunk 1 above)
Comment (dbkr) [reply src/components/views/elements/AppTile.js]: Yeah, was on the fence about whether to do this now or leave it until moving more stuff to the store, but probably better to do it now.

**Comment by bwindels** [comment]
@dbkr Test failed before on:
```
     TypeError: Cannot read property 'on' of null
    at ActiveWidgetStore.start (test/src/stores/ActiveWidgetStore.js:51:30 <- test/all-tests.js:174459:45)
```
Probably still do

**Pull request merged**